src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -172,10 +172,16 @@ export const UnderlineNavItem = forwardRef(
                 {children}
               </Box>
             )}
-            {counter && (
+            {loadingCounters ? (
               <Box as="span" data-component="counter" sx={counterStyles}>
-                {loadingCounters ? <LoadingCounter /> : <CounterLabel>{counter}</CounterLabel>}
+                <LoadingCounter />
               </Box>
+            ) : (
+              counter !== undefined && (
```

Comment:
Context: the type of the `counter` is `string | number` and not every item has to have a counter.